elastic/kibana
# [Snapshot & Restore] Fix repository types on Cloud

## Summary
Fixes https://github.com/elastic/kibana/issues/125424
This PR fixes an issue when module repository types were not displayed on Cloud when registering a repository. These repository types were converted from plugins and now are available in Elasticsearch out of the box (see https://github.com/elastic/elasticsearch/pull/81870). 

This PR also adds an api integration test to check for default repo types both on prem and on Cloud. But it's currently not possible to run functional tests against my branch on Cloud using [Cloud flaky tests runner](https://internal-ci.elastic.co/view/Stack%20Tests/job/elastic+estf-cloud-kibana-flaky-test-runner/347/). I will re-run the tests once this PR is merged and the changes are in 8.2.0-SNAPSHOT. 


### How to test
1. Navigate to Stack Management -> Snapshot and Restore
2. Switch to "Repositories" tab
3. Click "Register repository" button 
a. On Cloud 3 repository types are available (azure, google cloud and s3)
b. On prem there are additionally system file and url repo types 
NOTE: to test on Cloud see "Build and Deploy to Cloud" details in the build information below. 

### Checklist


## Release Note
Registering a snapshot repository is again possible in Snapshot and Restore app.
